modules/031-linstor/images/linstor-pools-importer/main.go

```diff
@@ -0,0 +1,386 @@
+	klog.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+}
+
+func main() {
+	delaySeconds := flag.Int("delay", 10, "Delay in seconds between scanning attempts")
+
+	flag.Parse()
+	printVersion()
+
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+
+	if nodeName == "" {
+		klog.Fatalln("Required NODE_NAME env variable is not specified!")
+	}
+	klog.Infof("nodeName: %+v", nodeName)
+	if podName == "" {
+		klog.Fatalln("Required POD_NAME env variable is not specified!")
+	}
+	klog.Infof("podName: %+v", podName)
+	if namespace == "" {
+		klog.Fatalln("Required NAMESPACE env variable is not specified!")
+	}
+	klog.Infof("namespace: %+v", namespace)
+
+	ctx := context.Background()
+
+	// Get a config to talk to the apiserver
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		klog.Errorln("Failed to get kubernetes config:", err)
+	}
+
+	kc, err := kclient.New(config, kclient.Options{})
+	if err != nil {
+		klog.Fatalln("Failed to create Kubernetes client:", err)
+	}
+
+	var pod v1.Pod
+	err = kc.Get(ctx, types.NamespacedName{Name: podName, Namespace: namespace}, &pod)
+	if err != nil {
+		klog.Fatalf("look up owner(s) of pod %s/%s: %v", namespace, podName, err)
+	}
+	owner := v1.ObjectReference{
+		APIVersion: "v1",
+		Kind:       "Pod",
+		Name:       pod.GetName(),
+		Namespace:  pod.GetNamespace(),
+		UID:        pod.GetUID(),
+	}
+	klog.Infof("using %s/%s as owner for Kubernetes events", owner.Kind, owner.Name)
+
+	lc, err := lclient.NewClient()
+	if err != nil {
+		klog.Fatalln("failed to create LINSTOR client:", err)
+	}
+
+	klog.Infof("Starting main loop (delay: %d seconds)", *delaySeconds)
+	candidatesChannel := runCandidatesLoop(ctx, getCandidates, time.Duration(*delaySeconds)*time.Second)
+
+	for storagePool := range candidatesChannel {
+		klog.Infof("Got %s candidate: %+v\n", storagePool.ProviderKind, storagePool)
+		_, err = lc.Nodes.Get(ctx, nodeName)
+		if err != nil {
+			klog.Fatalln("Failed to get LINSTOR node", err)
+		}
+		_, err = lc.Nodes.GetStoragePool(ctx, nodeName, storagePool.StoragePoolName)
+		if err != nil && err == lclient.NotFoundError {
+			err = lc.Nodes.CreateStoragePool(ctx, nodeName, storagePool)
+			if err != nil {
+				sendKubernetesEvent(ctx, kc, &owner, v1.EventTypeWarning, "Failed", "Failed to create LINSTOR storage pool"+err.Error())
+				klog.Fatalln("Failed to create LINSTOR storage pool", err)
+			}
+			sendKubernetesEvent(ctx, kc, &owner, v1.EventTypeNormal, "Created", "Created LINSTOR storage pool: "+nodeName+"/"+storagePool.StoragePoolName)
+		} else if err != nil {
+			klog.Fatalln("Failed to get LINSTOR storage pool", err)
+		}
+
+		// Get the maximum number of available replicas
+		opts := lclient.ListOpts{StoragePool: []string{storagePool.StoragePoolName}}
+		sp, err := lc.Nodes.GetStoragePoolView(ctx, &opts)
+		if err != nil {
```

Comment:
> `opts`: how can we ensure we don't miss pools due to pagination?

Nice catch! I think we will never be affected by this since we need just three, But I think we can add limit option that equals to `maxReplicasNum` (3) 
 
https://app.swaggerhub.com/apis-docs/Linstor/Linstor/1.12.0#/developers/viewStoragePools